src/js/components/awardv2/idv/activity/chart/ActivityChart.jsx

```diff
@@ -134,12 +135,30 @@ export default class ActivityChart extends React.Component {
             if (this.props.showTooltipStroke && (this.props.awardIndexForTooltip === index)) {
                 style = { stroke: '#3676b6', strokeWidth: 1 };
             }
+            // handle overspending
+            let overspentStyle = null;
+            let overspentWidth = null;
+            if (bar._obligatedAmount > bar._awardedAmount) {
+                overspentStyle = { fill: "url(#diagonalHatch)" };
+                overspentWidth = width;
+            }
             return (
                 <g
                     tabIndex="0"
                     className="activity-chart-bar-container"
                     key={`bar-${bar._awardedAmount}-${index}`}
                     description={description}>
+                    <defs>
+                        <pattern
+                            id="diagonalHatch"
+                            width="10"
+                            height="10"
+                            patternTransform="rotate(135 0 0)"
+                            patternUnits="userSpaceOnUse">
+                            <rect width="100%" height={barHeight} fill="#94BFA2" />
+                            <rect width="2" height={barHeight} fill="rgb(188,92,35)" />
+                        </pattern>
+                    </defs>
```

Comment:
This fn should be about 3 fns. Its way too big IMO.

Reply:
@MaxwellKendall moved x and y ranges out to their own functions and moved the graph dimensions to their own functions. Any other suggestions on breaking this down?